src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -910,5 +948,99 @@ private ICollection<ProjectWithInnerNodes> LoadProjects(IEnumerable<ProjectGraph
             // Just return the projects not the whole dictionary as it was just used to group the projects together
             return projects.Values;
         }
+
+        /// <summary>
+        /// It evaluates the project and returns true if the project has CentralPackageVersionManagement enabled.
+        /// </summary>
+        /// <param name="project">The <see cref="IMSBuildProject"/> for which the CentralPackageVersionManagement will be evaluated.</param>
+        /// <param name="log">A <see cref="ILogger"/> instance.</param>
+        /// <returns>True if the project has CentralPackageVersionManagement enabled.</returns>
+        internal static bool IsCentralVersionsManagementEnabled(IMSBuildProject project, Common.ILogger log)
+        {
+            var projectStyleResult = BuildTasksUtility.GetProjectRestoreStyle(
```

Comment:
The project style is now calculated twice.  I would move up the original call in `GetPackageSpec` and pass that into this method rather than calculate it again.  That would also probably make this method not necessary.

Reply:
Refactored to avoid MSBuild property retrieval duplication.